### PR TITLE
visualization of progression of fits as videos

### PR DESCRIPTION
**Pull request by palday**
I wrote this mostly to test out creating animations with Makie, but it also really highlights how quickly the nonlinear optimizer gets within the neighborhood of the optimum.

In this video, you can see how the initial steps are a bit coarse and the LL is all over, then there are several steps in the right direction and the LL quickly converges. There are some changes in a parameters as you get further out.

https://user-images.githubusercontent.com/1677783/129115201-6a485e1b-367f-4007-88da-6c78c1b3920e.mp4

~(Maybe I should color code the theta components to indicate which are variances and which are correlations.)~ (The components are colored by blocking variable.)

Same here:

https://user-images.githubusercontent.com/1677783/129115178-f9d2c9c6-5c18-414a-ac74-d09da627f7da.mp4

In these videos, you can see that giant jump in the likelihood as the moment when shrinkage kicks in quite hard and the stars collapse: 

https://user-images.githubusercontent.com/1677783/129115318-23d44aef-c669-4416-a285-0f4f780ddd4d.mp4

https://user-images.githubusercontent.com/1677783/129115344-6e4fb257-6970-4dfd-bfd3-4db273268567.mp4

(This definitely isn't the most efficient way to generate that animation since it winds up re-evaluating each step that the optimizer took.)

I would be quite keen to do another of these with the gradient plotted. I thought about plotting the change in theta between iterations as the L2 norm of the difference vector or maybe even just the cosine distance that the machine-learning folks love. 

I would also like to do this on a very large simulated dataset that's as far from singular as we can get, but that will require a fair amount of computer time.

### CODE REVIEW

File: notebooks/vizfit.jl [hunk 1]
```diff
@@ -0,0 +1,1410 @@
...
+
+# ╔═╡ 42726a44-4723-4d9f-954e-79635ea0d365
+llvid = let path = path, fig = Figure(), frate = frate, l = l
+
+	pp = Node(Point2f0[])
+    ax = Axis(fig[1, 1])
+	
+	lines!(ax, pp)
+	ax.xlabel = "Iteration"
+	ax.ylabel = "Log likelihood"
```
Comment (palday) [inline notebooks/vizfit.jl:176]:
```suggestion
	ax.ylabel = "Deviance"
```

File: notebooks/vizfit.jl [hunk 2]
```diff
@@ -0,0 +1,1410 @@
...
+# ╔═╡ 7a69147b-0b13-4339-9d53-9748ed506c03
+LocalResource(llvid)
+
+# ╔═╡ fc29e293-8eab-4f3b-99c3-4ae11389f881
+θvid = let path = path, fig = Figure(), frate = frate, l = l
+	
+	ppll = Node(Point2f0[])
+    axll = Axis(fig[1, 1])
+	lines!(axll, ppll)
+	axll.ylabel = "Log likelihood"
```
Comment (palday) [inline notebooks/vizfit.jl:194]:
```suggestion
	axll.ylabel = "Deviance"
```

**Review by dmbates — APPROVED** [review]
This is fantastic.  Thanks!  Enthusiastically approve.

**Comment by dmbates** [comment]
Do you think we add a general option to the `fit` methods to preserve the trial parameter values and objective values?  Should we use a vector of tuples instead of a vector of vectors for the parameter values?

**Comment by palday** [comment]
@dmbates very good question. I think adding an extra method where you pass a vector for logging is relatively straightforward and worthwhile (say with kwarg `fitlog=nothing` and then branching on `isnothing(fitlog)`), but then technically that method is mutating, so we should document that very well. As to whether tuples or static vectors or ordinary vectors would be best fort internal storage ... probably static vectors because they tend to play nicer with the plotting ecosystems (as subtypes of `AbstractArray`).  Static vectors are also what we use for storing the bootstrap samples.

**Comment by dmbates** [comment]
Agree on both points: kwarg of `fitlog` with a default of `nothing` and the use of static vectors.

**Pull request merged**